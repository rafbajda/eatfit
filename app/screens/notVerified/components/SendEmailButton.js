import React from 'react';
import { Row, Text } from 'native-base';
import styles, { OtherActionButton } from '../styles/notVerifiedStyles';

const SendEmailButton = props => {
    const { sendEmail, OtherActionTextStyle, OtherActionRowStyle } = { ...props, ...styles };
    return (
        <Row style={OtherActionRowStyle}>
            <OtherActionButton onPress={() => sendEmail()} rounded>
                <Text style={OtherActionTextStyle}>Send another email</Text>
            </OtherActionButton>
        </Row>
    );
};

export default SendEmailButton;
