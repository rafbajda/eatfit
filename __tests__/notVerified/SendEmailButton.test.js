import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import SendEmailButton from "../../app/screens/notVerified/components/SendEmailButton";

describe('SendEmailButton snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it('renders component', async () => {
        const tree = renderer.create(<SendEmailButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
