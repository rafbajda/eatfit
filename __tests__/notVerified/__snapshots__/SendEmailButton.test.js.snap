// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendEmailButton snapshot renders component 1`] = `
<View
  style={
    Object {
      "flex": 0,
      "flexDirection": "row",
      "height": "auto",
      "justifyContent": "center",
      "paddingBottom": 12,
      "paddingTop": 12,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#007aff",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 57,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          Object {
            "fontSize": 16,
            "textAlign": "center",
          },
        ]
      }
      uppercase={false}
    >
      [missing "en.buttons.sendAnotherEmail" translation]
    </Text>
  </View>
</View>
`;
